cpp/src/gis/point_in_polygon.cu

```diff
@@ -0,0 +1,184 @@
+}
+
+/** 
+ *  @brief Determine whether or not coordinates (query points) are completely inside a static polygon
+ *
+ * @param[in] poly_lats: Pointer to latitudes of a polygon
+ * @param[in] poly_lons: Pointer to longitudes of a polygon
+ * @param[in] point_lats: Pointer to latitudes of many query points
+ * @param[in] point_lons: Pointer to longitudes of many query points
+ * @param[in] poly_size: Size of polygon
+ * @param[in] point_size: Total number of query points
+ * @param[out] point_is_in_polygon: Pointer indicating if the i-th query point is inside or not
+ *
+ * @returns
+ */
+ template <typename T>
+ __global__ void point_in_polygon_kernel(const T* poly_lats,
+                                    const T* poly_lons,
+                                    const T* point_lats,
+                                    const T* point_lons,
+                                    gdf_size_type poly_size,
+                                    gdf_size_type point_size,
+                                    int8_t* point_is_in_polygon)
+{
+    gdf_index_type start_idx = blockIdx.x * blockDim.x + threadIdx.x;
+ 
+    for(gdf_index_type idx = start_idx; idx < point_size; idx += blockDim.x * gridDim.x)
+    {
+        T point_lat = point_lats[start_idx];
+        T point_lon = point_lons[start_idx];
+        gdf_size_type count = 0;
+ 
+        for(gdf_index_type poly_idx = 0; poly_idx < poly_size - 1; poly_idx++) 
+        {
+            if(poly_lons[poly_idx] <= point_lon && point_lon < poly_lons[poly_idx + 1])
+            {
+                if (orientation(poly_lons[poly_idx], poly_lats[poly_idx], poly_lons[poly_idx + 1], poly_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                {
+                    count++;
+                }
+            }
+            else if (point_lon <= poly_lons[poly_idx] && poly_lons[poly_idx + 1] < point_lon) 
+            {
+                if (orientation(poly_lons[poly_idx], poly_lats[poly_idx], poly_lons[poly_idx + 1], poly_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                {
+                    count++;
+                }
+            }
+        }
+        if ((count > 0) && (count % 2 == 0)) point_is_in_polygon[start_idx] = 1;
+        else point_is_in_polygon[start_idx] = 0;
+    }
+}
+
+struct point_in_polygon_functor {
+    template <typename col_type>
+    static constexpr bool is_supported()
+    {
+        return std::is_arithmetic<col_type>::value ;
+    }
+ 
+    template <typename col_type, typename std::enable_if< is_supported<col_type>() >::type* = nullptr>
+    void operator()(gdf_column const & d_poly_lats, gdf_column const & d_poly_lons,
+                    gdf_column const & d_point_lats, gdf_column const & d_point_lons,
+                    gdf_column const & d_point_is_in_polygon)
+    {
+
+        gdf_size_type min_grid_size = 0, block_size = 0;
+        CUDA_TRY( cudaOccupancyMaxPotentialBlockSize(&min_grid_size, &block_size, point_in_polygon_kernel<col_type>) );
```

Comment:
Are you sure you want to use `min_grid_size` here? It might give higher performance to have more blocks per SM running than the minimum. In the case where blocks run for different amounts of time, you might suffer a poor tail effect if you only use the minimum. Have you benchmarked? https://devblogs.nvidia.com/cuda-pro-tip-minimize-the-tail-effect/

Reply:
What you are saying is something like .. I should before compare with the needed gridsize, like that: 

```
gdf_size_type min_grid_size = 0, block_size = 0
CUDA_TRY( cudaOccupancyMaxPotentialBlockSize(&min_grid_size, &block_size, point_in_polygon_kernel<col_type>) );

// find needed gridsize
gdf_size_type grid_size = (d_point_lats.size + blocksize - 1) / blocksize;
```
Then launch the kernel
`point_in_polygon_kerel<col_type> <<< grid_size, block_size >>> (...)`

Reply:
Yep. But use `cudf::util::cuda::grid_config_1d grid{d_point_lats.size, blocksize, 1};`, which protects against overflow in the rounding calculation.